internal/v1/handler_compose_image.go

```diff
@@ -654,8 +659,53 @@ func (h *Handlers) buildCustomizations(ctx echo.Context, cust *Customizations, s
 	}
 
 	if cust.Openscap != nil {
-		res.Openscap = &composer.OpenSCAP{
-			ProfileId: cust.Openscap.ProfileId,
+		profile, err := cust.Openscap.AsOpenSCAPProfile()
+		if err != nil {
+			return nil, err
+		}
+
+		if profile.ProfileId != "" {
+			res.Openscap = &composer.OpenSCAP{
+				ProfileId: profile.ProfileId,
+			}
+		}
+		policy, err := cust.Openscap.AsOpenSCAPCompliance()
+		if err != nil {
+			ctx.Logger().Error(err.Error())
+			return nil, err
+		}
+
+		if policy.PolicyId != uuid.Nil {
+
+			if !unleash.Enabled(unleash.CompliancePolicies) {
+				return nil, echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Feature %s not enabled", string(unleash.CompliancePolicies)))
+			}
+
+			pdata, err := h.server.complianceClient.PolicyDataForMinorVersion(ctx.Request().Context(), 8, 10, policy.PolicyId.String())
```

Comment:
Harcoded version (8.10) reminder. Was this intentional?

Reply:
Oh woops :facepalm:  no this should be parsed from the distro.